app/controllers/api/v1/merchant_items_controller.rb

```diff
@@ -2,7 +2,7 @@ class Api::V1::MerchantItemsController < ApplicationController
   rescue_from ActiveRecord::RecordNotFound, with: :not_found_response
 
   def show
-    item = Item.find_by(id: params[:id])
+    item = Item.find(params[:id])
```

Comment:
Great fix to the error handling working correctly now, changing this from find_by to find.